core/Tracker/Request.php

```diff
@@ -198,6 +198,19 @@ public static function authenticateSuperUserOrAdminOrWrite($tokenAuth, $idSite)
             return false;
         }
 
+        // Now checking the list of admin token_auth cached in the Tracker config file
+        if (!empty($idSite) && $idSite > 0) {
+            $website = Cache::getCacheWebsiteAttributes($idSite);
+            $userModel = new \Piwik\Plugins\UsersManager\Model();
+            $tokenAuth = $userModel->hashTokenAuth($tokenAuth);
```

Comment:
Guess you need to use another variable name here, as it otherwise will overwrite the tokenAuth used for authentication below. Guess that's the reason for all the test failues...